pkg/ipamd/ipamd.go

```diff
@@ -826,8 +826,11 @@ func (c *IPAMContext) tryAllocateENI(ctx context.Context) error {
 
 		log.Infof("ipamd: using custom network config: %v, %s", eniCfg.SecurityGroups, eniCfg.Subnet)
 		for _, sgID := range eniCfg.SecurityGroups {
-			log.Debugf("Found security-group id: %s", sgID)
-			securityGroups = append(securityGroups, aws.String(sgID))
+			// skip null string - issue 1626
+			if len(sgID) > 0 {
```

Comment:
We shouldn't be adding such check either for nil SG or invalid SG since this is a user induced error and needs correction. Instead we can improve the logging and give a chance for user to fix this. Maybe in future we can add this as part of node/pod events?